contributors/design-proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,405 @@
+# Alternate representations of API resources
+
+## Abstract
+
+Naive clients benefit from allowing the server to returning resource information in a form
+that is easy to represent or is more efficient when dealing with resources in bulk. It
+should be possible to ask an API server to return a representation of one or more resources
+of the same type in a way useful for:
+
+* Retrieving a subset of object metadata in a list or watch of a resource, such as the
+  metadata needed by the generic Garbage Collector or the Namespace Lifecycle Controller
+* Dealing with generic operations like `Scale` correctly from a client across multiple API
+  groups, versions, or servers
+* Return a simple tabular representation of an object or list of objects for naive
+  web or command-line clients to display (for `kubectl get`)
+* Return a simple description of an object that can be displayed in a wide range of clients
+  (for `kubectl describe`)
+* Return the object with fields set by the server cleared (as `kubectl export`)
```

Comment:
List of field has to be on the server side and vary by type.  We have lots of cases where base export is insufficient (the context of what scope to extract - namespace, cluster, version of Kube needs to be also provided).